AffirmSDK/AffirmItem.h

```diff
@@ -67,8 +72,27 @@ NS_ASSUME_NONNULL_BEGIN
 NS_SWIFT_NAME(item(name:sku:unitPrice:quantity:url:));
 
 /**
- Initializer. See properties for more details.
+ Convenience constructor. See properties for more details.
+ 
+ @param name Item name.
+ @param SKU Item SKU.
+ @param unitPrice Price per item.
+ @param quantity Number of items purchased.
+ @param URL URL of the item.
+ @param categories An array of lists that indicate the various categories that apply to this product, and the hierarchy of those category definitions.
+ @return The newly created item.
+ */
++ (AffirmItem *)itemWithName:(NSString *)name
+                         SKU:(NSString *)SKU
+                   unitPrice:(NSDecimalNumber *)unitPrice
+                    quantity:(NSInteger)quantity
+                         URL:(NSURL *)URL
+                  categories:(nullable NSArray<NSArray<NSString *> *> *)categories
```

Comment:
I'm coming into this cold and don't have context. But an array of arrays is kind of weird to wrap one's head around.

Would it be clearer to define an `AffirmCategory` like:

```obj-c
@interface AffirmCategory
@property (nonatomic, nonnull) NSString name;
@property (nonatomic, nullable) NSArray<NSString*> subCategories;
... etc.
```

and have `categories` be an `NSArray<AffirmCategory*>*`?

Reply:
Right now, both the backend and Android SDK use a two-dimensional string array without any subdivisions. Do we need to standardize this? If we decide to define an AffirmCategory, it only needs to include an array, the name isn't really necessary.

Backend:
https://docs.affirm.com/developers/reference/the-item-object

Android SDK:
https://github.com/Affirm/affirm-merchant-sdk-android/pull/137/files#diff-0f9e2c64c69d37cfd2646df60aecfb5bf5ce6eef5cbee13c2981134d38b47a0eR54

Reply:
Due to some issues with GitHub Actions on the latest macOS, downgrading to macOS-12 can resolve the problem. You can find more details here: https://github.com/actions/runner-images/issues/9591.

Reply:
Ed's comment seems fine to me, might help with clarity.
Will we run into any issues in the future with this macOS "downgrade"?

Reply:
Sure, I'll optimize it. The downgrade does not impact the SDK, it simply enables tests to run normally.

Reply:
@eb-smith-affirm @bf-affirm It has been optimized, thanks for your response.